client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/InitializeParams.java

```diff
@@ -41,11 +42,18 @@ public class InitializeParams {
   private final boolean enableSecurityHotspots;
   private final List<SonarQubeConnectionConfigurationDto> sonarQubeConnections;
   private final List<SonarCloudConnectionConfigurationDto> sonarCloudConnections;
+  private final String sonarlintUserHome;
 
-  public InitializeParams(Path storageRoot, Set<Path> embeddedPluginPaths, Map<String, Path> connectedModeExtraPluginPathsByKey,
+  /**
+   * @param productKey        SonarLint product key (vscode, idea, eclipse, ...)
```

Comment:
It's not clear to me if this value could be any random unique identifier, or if there is a list of predefined values that are accepted.

Reply:
At the moment, this value is only used in telemetry. Maybe I should rename it `telemetryProductKey`?